File: Embassat/View/Menu/CADEMMenuCollectionViewCell.m
//
//  CADEMMenuCollectionViewCell.m
//  Embassat
//
//  Created by Joan Romano on 09/03/14.
//  Copyright (c) 2014 Crows And Dogs. All rights reserved.
//

#import "CADEMMenuCollectionViewCell.h"

#import "UIColor+EMAdditions.h"

@interface CADEMMenuCollectionViewCell ()

@property (nonatomic, weak) IBOutlet UILabel *optionNameLabel;

@end

@implementation CADEMMenuCollectionViewCell

- (void)setupView
{
    RAC(self.optionNameLabel, text) = RACObserve(self, optionName);
    
    self.optionNameLabel.font = [UIFont em_titleFontOfSize:30.0f];
    self.optionNameLabel.adjustsFontSizeToFitWidth = YES;
    
    RAC(self, optionNameLabel.textColor) =
    [RACSignal
     combineLatest:@[RACObserve(self, selected), RACObserve(self, highlighted)]
     reduce:^id(NSNumber *selected, NSNumber *highlighted){
         return selected.boolValue || highlighted.boolValue ? [UIColor em_barTintColor] : [UIColor whiteColor];
     }];
}

@end

File: Embassat/View/Menu/CADEMMenuCollectionViewCell.swift
//
//  CADEMMenuCollectionViewCell.swift
//  Embassa't
//
//  Created by Joan Romano on 25/05/15.
//  Copyright (c) 2015 Crows And Dogs. All rights reserved.
//

import UIKit

public class CADEMMenuCollectionViewCell: CADRootCollectionViewCell {
    
    @IBOutlet weak var optionNameLabel: UILabel?
    public var optionName: String = "" {
        didSet {
            optionNameLabel?.text = optionName
        }
    }
    
    override public var selected: Bool {
        didSet {
            if selected {
                optionNameLabel?.textColor = UIColor.em_barTintColor()
            } else {
                optionNameLabel?.textColor = UIColor.whiteColor()
            }
        }
    }
    
    override public var highlighted: Bool {
        didSet {
            if highlighted {
                optionNameLabel?.textColor = UIColor.em_barTintColor()
            } else {
                optionNameLabel?.textColor = UIColor.whiteColor()
            }
        }
    }
    
    public override func setupView() {
        super.setupView()
        
        optionNameLabel?.font = UIFont.em_titleFontOfSize(30.0)
        optionNameLabel?.adjustsFontSizeToFitWidth = true
    }
}
